### ISSUE TITLE
Please get the 9.0 Proton tree (and bleeding-edge) working

### ISSUE THREAD

**Issue by KyunLFA** [Frogging-Family/wine-tkg-git#1175]
Currently on my end (CachyOS), building with _proton_branch set to anything higher than the experimental_8.0 tree in proton-tkg (and vrclient also overrides to an even older commit) with proton flavor set to Wine master produces an obscene amount of errors I could not fix try as I may, unrelated to wine itself as it builds just fine, but instead occurring in the Proton bits like vrclient and lsteamclient (that aren't addressed by silencing the implicit declaration and incompatible pointer errors).

At this point I'm nudging towards the idea that it would be easier to just use the official proton building script (but somehow beating the requirement for containers) because so much has changed since 8.0-experimental.

Please consider fixing proton-tkg with 9.0 and bleeding-edge branches if you find that they are also broken for you.

Thanks!

**Comment by Tk-Glitch** [comment]
That tree you mention is only used on our side for the steam-specific tooling (lsteamclient, vrclient and steam_helper) and indeed it's totally broken past a certain commit and not compatible with our build system. The wine part doesn't depend on it and isn't defined by that variable. Other external tooling is handled through their own respective trees, also not depending on those steam-specific libs/tools.
We are overriding the tooling tree commit to f5e9c76903e4e18e0416e719a6d42d0cb00998aa with https://github.com/Frogging-Family/wine-tkg-git/blob/master/proton-tkg/proton-tkg.sh#L968 as any newer commit depends on that different approach we do not support. However unless you nuked that part of the script it should use that commit and work just fine (as it does for me and the CI).
If you haven't fiddled with it, the error might be unrelated and just a dependency issue (in that case please attach a log or a copy of those errors you get).

**Comment by KyunLFA** [comment]
Yes I have overridden to use the newer proton tree (deleted the override that vrclient does) to see if it would work, and it's severely broken when doing that.

My intention was to ask for newer Proton tree support when you find the time to add it.

Thanks for answering

**Comment by Tk-Glitch** [comment]
There's actually no vrclient specific target commit, we're using what's in the tree on f5e9c769, while using openvr master branch (which is newer than what's used in bleeding edge currently).
So, new question from me. What's the reason for you to want/need that newer tooling? VR I assume (that would be the only point to it really)?

Anyway it's being worked on but I have no ETA yet.

**Comment by KyunLFA** [comment]
I guess for same reason we all update, to be on the bleeding edge and the perks it gives, I would personally prefer if the whole Proton tree could be used on the latest commit, as minor as that may be, so that it matches what's currently available with the official Proton 9.0, which could be seen as wine-tkg being slightly "behind" in that aspect (mostly the lsteamclient and steam_helper updates). Sorry if this seems stupid/weird to you.

No biggie with the ETA subject, I just wanted to know if it's being developed or not, no rushing intended.

Thanks!

**Comment by Tk-Glitch** [comment]
Right now it's not effectively offering more "perks" and is more of an integration change, which lead us to this fallback situation that was deemed acceptable as a buffer. The steam.exe and lsteamclient triggers are happening on game launch basically and we're done. No expected user-side change.
There are other plans regarding future proton handling, too, which I'll hopefully get to a satisfying spot soon and will offer more options. I'll update this ticket when we have proper handling of the new compilation process for the steam bridge and friends.

**Comment by KyunLFA** [comment]
@Tk-Glitch It _seems_ this became a real issue, **as far as I can tell**: Dragon Ball Sparking Zero does not launch with Proton-TKG (Wine Master and Valve XBE), even the prebuilt ones and even when also testing with Steam's runtime, with the log containing the entries: `err:lsteamclient:init_ntdll_so_funcs NtQueryVirtualMemory status 0xc0000135.`, and `12428.932:013c:0140:err:steamclient:create_win_interface Don't recognize interface name: SteamClient021`.  It does launch with normal Proton, GE and CachyOS versions as well, which use the "standard" packaging and use the latest steam bits.

Would you please revisit this issue? (sorry for a ping on a Sunday)
